apis/tempo/v1alpha1/microservices_types.go

```diff
@@ -80,11 +80,38 @@ type MicroservicesSpec struct {
 	ReplicationFactor int `json:"replicationFactor,omitempty"`
 }
 
+// ConditionStatus defines the status of a condition (e.g. ready or degraded).
+type ConditionStatus string
+
+const (
+	// ConditionReady defines that all components are ready.
+	ConditionReady ConditionStatus = "Ready"
+	// ConditionDegraded defines that one or more components are in a degraded state.
+	ConditionDegraded ConditionStatus = "Degraded"
+)
+
+// ConditionReason defines possible reasons for each condition.
+type ConditionReason string
+
+const (
+	// ReasonReady defines a healthy tempo instance.
+	ReasonReady ConditionReason = "Ready"
+	// ReasonInvalidStorageConfig defines that the object storage configuration is invalid (missing or incomplete storage secret).
```

Comment:
Just asking: would it make sense to differentiate between the two conditions? To provide more information in the log or something

Reply:
I don't follow, can you elaborate on that? Currently the condition is `Ready` if everything is working, and `Degraded` if something is broken. If it's `Degraded`, the `ConditionReason` will explain why it's `Degraded`.

Reply:
I see. I was asking because, from the description of the status, it seems like there is actually more than one condition for the "Degraded" state:
> missing or incomplete storage secret

So, I was wondering if it would be a good idea to have something like `Missing` and `Incomplete Secret` instead of just `Degraded`.

Reply:
Ah, now I understand (I thought "differentiate between the two conditions" was referring to `Ready` vs. `Degraded`, which got me confused).

> So, I was wondering if it would be a good idea to have something like `Missing` and `Incomplete Secret` instead of just `Degraded`.

Good question - incomplete secret could also have multiple reasons (`MissingEndpoint`, `InvalidEndpoint`, `MissingAccessKeyId`, `MissingAccessKeySecret`, etc).

In my opinion the `Reason` field is most useful if another operator reads this field and can react based on it (in this sense, I'd count it as part of the API which the tempo-operator exposes). In this case however I guess it'll be only administrators reacting to this error condition. Admins then (hopefully) also read the full error message (which gives further details).